pkgs/development/compilers/llvm/13/bintools/default.nix

```diff
@@ -1,11 +1,17 @@
-{ runCommand, stdenv, llvm, lld, version }:
+{ runCommand, stdenv, llvm, lld, version, lib }:
 
 let
   prefix =
     if stdenv.hostPlatform != stdenv.targetPlatform
     then "${stdenv.targetPlatform.config}-"
     else "";
-in runCommand "llvm-binutils-${version}" { preferLocalBuild = true; } ''
+in runCommand "llvm-binutils-${version}" {
+  preferLocalBuild = true;
+  passthru = {
+    isLLD = true;
```

Comment:
I'm a bit torn on this one. I think it's alright and probably useful, *but* it should be introduced in a consistent manner.

My proposal would be to add this separately bundled with a similar disambiguator added for `cctools` and GNU `binutils` as well. `bintools-wrapper` should also add the missing disambiguators via elaboration.

For now `stdenv.hostPlatform.linker == "lld"` works just as well, with the slight disadvantage of inaccuracy if a non-default `bintools` is passed *manually*.

Reply:
> My proposal would be to add this separately bundled with a similar disambiguator added for cctools and GNU binutils as well. bintools-wrapper should also add the missing disambiguators via elaboration.

I don't object to this as an end, but do we need to do it all at once?

> For now stdenv.hostPlatform.linker == "lld" works just as well

No it doesn't, as I've been testing stdenvs using lld which is not the default for the platform.

Reply:
Doing all things at once was kind of my point. E.g. `clangUseLLVM` doesn't really work well outside of a platform that actually has `useLLVM = true` and  `linker = "lld"` which is why it's used nowhere in nixpkgs except in the cross stdenv. In fact all LLVM stdenvs currently have this sometimes inconvenient property of respecting the platform to an extent, see this hack in firefox: https://github.com/NixOS/nixpkgs/blob/879951577d17a74f69a44e9f67f3fb65c31b9c03/pkgs/applications/networking/browsers/firefox/common.nix#L151-L153

 So arguably this PR is two things at once: adding windows llvm cross and making compiler-rt / lld detecteable independently from the platform (i.e. from the wrappers) at the same time.

Reply:
OK, so do we want an `isLLD`/`isBFD` etc. passthru on all the linkers, or `family = "lld"`, `family = "bfd"`, or something else?

Reply:
I guess currently it'd be:

* `isCctools`
* `isLld`
* `isGNU`
  * `isBfd`
  * `isGold`

Not sure if a string argument would be a good idea, so far we only have such things in the platform set.

wdyt @Ericson2314?

Reply:
That plan looks good to me. I am also undecided about that this strings; strings are nice to match `*Platform`, but `is*` handles "none of the above" nicely allowing "anonymous' options.